tvm/src/codegen/llvm/codegen_llvm.cc

```diff
@@ -1549,17 +1584,20 @@ void CodeGenLLVM::VisitStmt_(const KernelStmt* op) {
   }
   llvm::Function* f = module_->getFunction(op->name);
   builder_->CreateCall(f, arg_value);
-  llvm::BasicBlock* assert_true_kernel =
-      llvm::BasicBlock::Create(*ctx_, "assert_true_kernel", function_);
-  llvm::BasicBlock* assert_false_kernel =
-      llvm::BasicBlock::Create(*ctx_, "assert_false_kernel", function_);
-  llvm::Value* cond =
-      builder_->CreateLoad(llvm::Type::getInt32Ty(*ctx_), assert_global_ptr_);
-  builder_->CreateCondBr(cond, assert_true_kernel, assert_false_kernel);
-  builder_->SetInsertPoint(assert_false_kernel);
-  AssertFreeVars();
-  builder_->CreateRet(ConstInt32(0));
-  builder_->SetInsertPoint(assert_true_kernel);
+  if (kernel_has_assert_[op->name]) {
+    llvm::BasicBlock* assert_true_kernel =
+        llvm::BasicBlock::Create(*ctx_, "assert_true_kernel", function_);
+    llvm::BasicBlock* assert_false_kernel =
+        llvm::BasicBlock::Create(*ctx_, "assert_false_kernel", function_);
+    llvm::Value* assert_flag =
+        builder_->CreateLoad(llvm::Type::getInt32Ty(*ctx_), assert_global_ptr_);
+    llvm::Value* cond = builder_->CreateICmpEQ(assert_flag, ConstInt32(1));
+    builder_->CreateCondBr(cond, assert_true_kernel, assert_false_kernel);
+    builder_->SetInsertPoint(assert_false_kernel);
+    AssertFreeVars();
+    builder_->CreateRet(ConstInt32(1));
+    builder_->SetInsertPoint(assert_true_kernel);
+  }
```

Comment:
Why inserting the kernel only when there is an assertion inside? can you please explain?

Reply:
I did not modify this part, it is the same with what inside the master branch.

Reply:
@zzy82518996 can you run "git pull upstream master" to merge the latest changes in main branch? It seems that some changes are missing and the local CI/CD is not working as expected.